core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -407,12 +408,18 @@ public static void configureHadoopConf(Object maybeConfigurable, Object conf) {
    *
    * <p>The implementation must have a no-arg constructor.
    *
-   * @param impl full class name of a custom {@link MetricsReporter} implementation
+   * @param properties catalog properties which contains class name of a custom {@link
+   *     MetricsReporter} implementation
    * @return An initialized {@link MetricsReporter}.
    * @throws IllegalArgumentException if class path not found or right constructor not found or the
    *     loaded class cannot be cast to the given interface type
    */
-  public static MetricsReporter loadMetricsReporter(String impl) {
+  public static MetricsReporter loadMetricsReporter(Map<String, String> properties) {
+    if (!properties.containsKey(CatalogProperties.METRICS_REPORTER_IMPL)) {
```

Comment:
I prefer just calling `get` once and checking whether the value is null instead of doing two lookups (`containsKey` followed by `get`).

Reply:
PTAL